src/main/java/com/gorlah/kappabot/function/FunctionProcessor.java

```diff
@@ -14,8 +15,8 @@
 
     public String process(BotRequestMetadata metadata) {
         return botFunctions.stream()
-                .filter(botFunction -> botFunction.shouldProcess(metadata.getMessage()))
                 .map(botFunction -> botFunction.process(metadata))
+                .flatMap(Optional::stream)
```

Comment:
This is awesome and way cleaner.  👍

Reply:
Yeah. When I was playing around with stuff yesterday I found myself needing to change `BotFunction.shouldProcess` to take `BotRequestMetadata` instead of `String` because it needed additional information to decide whether to process or not. At that point I figured I might as well merge the boolean checking in with the actual processing and just return an `Optional` to indicate whether or not the request was accepted. It simplifies the `BotFunction` interface in a way that I think makes it better, and also prevents implementers of the interface from needing to consider what happens if `process` is called while `shouldProcess` is returning false.